banzai_nres/classify.py

```diff
@@ -58,6 +58,21 @@ def find_object_in_catalog(image, db_address):
 
 class StellarClassifier(Stage):
     def do_stage(self, image) -> NRESObservationFrame:
+
+        closest_previous_classification = dbs.get_closest_existing_classification(self.runtime_context.db_address,
+                                                                                  image.ra, image.dec)
+
+        if closest_previous_classification is not None:
+            previous_coordinate = SkyCoord(closest_previous_classification.ra, closest_previous_classification.dec,
+                                           unit=(units.deg, units.deg))
+            this_coordinate = SkyCoord(image.ra, image.dec, unit=(units.deg, units.deg))
+
+            # Short circuit if the object is already classified
+            if this_coordinate.separation(previous_coordinate) < 10.0 * units.arcsec:
```

Comment:
This will give incorrect results if e.g. a user is trying to observe both components of a close visual binary. Given the way that NRES acquisition currently works this is fine for now, but if we change the acquisition strategy to improve acquisition in these situations we will need to revisit this.

Reply:
I changed this to be the fiber size. So any resolved binaries should be ok.